plugins/connection/docker.py

```diff
@@ -214,7 +216,11 @@ def exec_command(self, cmd, in_data=None, sudoable=False):
         """ Run a command on the docker host """
         super(Connection, self).exec_command(cmd, in_data=in_data, sudoable=sudoable)
 
-        local_cmd = self._build_exec_cmd([self._play_context.executable, '-c', cmd])
+        if getattr(self._shell, "_IS_WINDOWS", False):
+            sudoable = False
+            local_cmd = self._build_exec_cmd(["cmd", "/c", cmd])
```

Comment:
I might be wrong but I believe using `cmd /c` here will mean any `raw` task with the connection plugin will be using `cmd` as the shell rather than PowerShell. You may want to change this based on whether the `ansible_shell_type` is either `cmd` or `powershell` to allow a user to change this as they desire.

Reply:
@jborean93 yes this is true. Bit when I change the code as follows I receive a `file not found`. So the interpreter (Powershell) cannot be found. 

![image](https://user-images.githubusercontent.com/14177833/117199844-f9cf5880-adea-11eb-85a1-026c7b3b1e05.png)

Reply:
Do you know what the default shell is for a Windows container, i.e. what runs when you do `docker exec ...`. Is it cmd or is it powershell.exe, or maybe no shell at all?

Basically right now if someone was to do the following

```yaml
- raw: echo "hi"
```

Is that going to be running under cmd or powershell? With the current changes here that will be running as `cmd` which might not be ideal if someone has set `ansible_shell_type=powershell`. Maybe the desire is to have `raw` with this connection plugin independent to the shell plugin used, if that's the case then the PR as it is makes sense. If you wish for this to be dependent on the shell plugin then you will have to hardcode `cmd.exe /c` when using the cmd shell and `powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command` for the powershell shell.

This might be something that @felixfontein may want to comment on.

Reply:
I don't think I can say something here. I've never used Docker with Windows containers, so I have zero experience what these contain.

Reply:
I mostly mean more in general, do you expect that `raw` should match the `ansible_shell_type` (powershell or cmd) or should it just be the same thing on the connection plugin?

Reply:
Matching `ansible_shell_type` would be nice I guess, though I'm not even sure whether Linux containers fully support that (I've never really looked at shell plugins). I also have no idea how complicated it would be to 'properly' implement this.